src/tribler/core/components/torrent_checker/torrent_checker/torrent_checker.py

```diff
@@ -114,67 +120,69 @@ async def check_random_tracker(self):
         except MalformedTrackerURLException as e:
+            session = None
             # Remove the tracker from the database
-            self.remove_tracker(tracker.url)
-            self._logger.error(e)
-            return False
+            self.tracker_manager.remove_tracker(url)
+            self._logger.warning(e)
 
+        if session is None:
+            self._logger.warning('A session cannot be created. The torrent check procedure has been cancelled.')
+            return
         # We shuffle the list so that different infohashes are checked on subsequent scrape requests if the total
         # number of infohashes exceeds the maximum number of infohashes we check.
         random.shuffle(infohashes)
         for infohash in infohashes:
             session.add_infohash(infohash)
 
-        self._logger.info("Selected %d new torrents to check on tracker: %s", len(infohashes), tracker.url)
+        self._logger.info(f"Selected {len(infohashes)} new torrents to check on tracker: {url}")
         try:
-            await self.connect_to_tracker(session)
-            return True
-        except:
-            return False
+            response = await self.get_tracker_response(session)
+        except Exception as e:  # pylint: disable=broad-except
+            self._logger.warning(e)
+        else:
+            self._logger.info(f'Random tracker check result: {response}')
+            self.update_health([response], 'random tracker')
 
-    async def connect_to_tracker(self, session):
+    async def get_tracker_response(self, session: TrackerSession) -> TrackerResponse:
+        self._logger.info(f'Connecting to the tracker: {session.tracker_url}')
         try:
-            info_dict = await session.connect_to_tracker()
-            return await self._on_result_from_session(session, info_dict)
+            response = await session.connect_to_tracker()
+            self._logger.info(f"Successfully connected to the tracker: {session.tracker_url}. Response: {response}")
+            return response
         except CancelledError:
-            self._logger.info("Tracker session is being cancelled (url %s)", session.tracker_url)
-            await self.clean_session(session)
+            self._logger.info(f"Tracker session is being cancelled: {session.tracker_url}")
         except Exception as e:
-            self._logger.warning("Got session error for URL %s: %s", session.tracker_url, str(e).replace('\n]', ']'))
-            await self.clean_session(session)
+            exception_str = str(e).replace('\n]', ']')
+            self._logger.warning(f"Got session error for the tracker: {session.tracker_url}\n{exception_str}")
             self.tracker_manager.update_tracker_info(session.tracker_url, False)
-            e.tracker_url = session.tracker_url
             raise e
+        finally:
+            await self.clean_session(session)
 
     @property
     def torrents_checked(self):
```

Comment:
The `load_torrents_checked_from_db` method loads all "recently checked" torrents, that is, torrents checked later than four hours ago from the previous Tribler run. As it turns out, due to some amazing combination of bugs that I'll describe in a separate issue, the database contains hundreds of "local" torrents with a "self_checked" flag set to `True`, a huge number of seeders and incorrect "future" last_check time. These torrents are also selected from the DB because "future" time is "later than four hours ago", so they are considered recently checked torrents. As a result, immediately after the Tribler launch PopularityCommunity starts spreading invalid health records to other peers.

To mitigate this, `load_torrents_checked_from_db` should not load invalid health objects with future dates from the DB, so the `last_check` value should be not only "more than four hours ago" but also "not in the future". And separately, invalid health records should be cleaned in the database.